kpatch-build/kpatch-build

```diff
@@ -167,10 +167,28 @@ find_core_symvers() {
 	[[ -e "$SYMVERSFILE" ]]
 }
 
+gcc_version_from_file() {
+	readelf -p .comment "$1" | grep -o 'GCC:.*'
+}
 gcc_version_check() {
+	local c="$TEMPDIR/test.c" aout="$TEMPDIR/test"
+	local out gccver kgccver
+
+	# gcc --version varies between distributions therefore extract version
+	# by compiling a test file and compare it to vmlinux's version.
+	echo 'void main(void) {}' > "$c"
+	out="$(gcc -pg -ffunction-sections -o "$aout" "$c" 2>&1)"
```

Comment:
I think it needs the `-c` option, so that it doesn't link in any of the GCC code for calling `main()`.  Otherwise on my F26 system I get two different versions of GCC in the `.comment` section:
```$ readelf -p .comment /tmp/a | grep -o 'GCC:.*'
GCC: (GNU) 7.1.1 20170622 (Red Hat 7.1.1-3)
GCC: (GNU) 7.2.1 20170915 (Red Hat 7.2.1-2)
```

Reply:
Interesting, I get only one version here (Debian). Will the version with `-c` match the version in vmlinux (which is also linked I guess)?

Reply:
Yes, compiling with `-c` matches the vmlinux GCC version.  Presumably because vmlinux is 100% compiled from source, whereas linking an executable brings in the `main()` boilerplate object code from somewhere.